### PR TITLE
Dom manipulation

### PR DESCRIPTION
**Pull request by MahtaMirzaei**
Closes: #54

### CODE REVIEW

File: mahta/JS/JS DOM-manipulation/script.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+function firstPage() {
+  document.body.innerHTML = "";
+  const buttonTag = document.createElement("button");
+  buttonTag.textContent = "Click Me!";
+  buttonTag.addEventListener("click", onButtonClickSet);
+
+  document.body.appendChild(buttonTag);
+  const elm = ["a", "b", "c", "d", "e", "g", "f", "h"];
+  elm.forEach((element, index) => {
+    elm[index] = document.createElement("div");
```
Comment (Mr-MRF-Dev) [inline mahta/JS/JS DOM-manipulation/script.js:73]: rewrite the code without using the index parameter.

**Review by Mr-MRF-Dev — CHANGES_REQUESTED** [review]

File: mahta/JS/JS DOM-manipulation/script.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  divElements[0].style.backgroundColor = aColor;
+  //Not functioning
+  subTree = [];
+  subTree.forEach((x) => {
+    if (Array.from(divElements[0].childNodes).includes(x)) {
+      x.style.backgroundColor = generateRandomColor(256, 256, 0);
+    } else {
+      x.style.backgroundColor = generateRandomColor(0, 0, 256);
+    }
+  });
```
[the remark this answers is not in the document]
Comment (MahtaMirzaei) [reply mahta/JS/JS DOM-manipulation/script.js:40]: it were supposed to generate yellow random color for a subtree and generate  blue random color for f subtree, and it failed.

File: mahta/JS/JS DOM-manipulation/script.js [hunk 3]
```diff
@@ -0,0 +1,116 @@
+function generateRandomColor(y, b) {
+  const yellow = Math.floor(Math.random() * (y - 155) + 155);
+  const blue = Math.floor(Math.random() * (b - 100) + 100);
+  return `rgb(${yellow}, ${yellow}, ${blue})`;
+}
```
Comment (Mr-MRF-Dev) [inline mahta/JS/JS DOM-manipulation/script.js:5]: It's okay. Additionally, you can use the `HSL` Color Code to generate a random color.

File: mahta/JS/JS DOM-manipulation/script.js [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+  };
+
+  const divElements = [];
+
+  Object.keys(elm).forEach((element) => {
+    const div = document.createElement("div");
+    div.innerText = element;
+    divElements.push(div);
+  });
+
```
Comment (Mr-MRF-Dev) [inline mahta/JS/JS DOM-manipulation/script.js:33]: To save divs in an object, use this syntax:

```js 
const allDivs = {};

  const divName = ["a", "b", "c", "d", "e", "g", "f", "h"];

  divName.forEach((name) => {
    allDivs[name] = document.createElement("div");
  });

```

**Review by Mr-MRF-Dev — CHANGES_REQUESTED** [review]

File: mahta/JS/JS DOM-manipulation/script.js [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+  divElements["f"].style.height = "250px";
+
+
+
+  const container = document.createElement("div");
+
+  document.body.appendChild(divElements["a"]);
+  divElements["a"].appendChild(divElements["b"]);
+  divElements["g"].appendChild(divElements["c"]);
+  divElements["C"].appendChild(divElements["d"]);
```
Comment (Mr-MRF-Dev) [inline mahta/JS/JS DOM-manipulation/script.js:44]: `C` or `c` ???

File: mahta/JS/JS DOM-manipulation/script.js [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+  divElements["b"].appendChild(divElements["g"]);
+  document.body.appendChild(divElements["f"]);
+  divElements["f"].appendChild(divElements["h"]);
+  divElements["f"].appendChild(divElements["i"]);
+  divElements["h"].appendChild(divElements["j"]);
+  divElements["a"].classList.add("container");
+  divElements["b"].classList.add("container");
+  divElements["C"].classList.add("container");
```
Comment (Mr-MRF-Dev) [inline mahta/JS/JS DOM-manipulation/script.js:52]: `C` or `c` ???

File: mahta/JS/JS DOM-manipulation/script.js [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+  divElements["g"].classList.add("container");
+  divElements["f"].classList.add("container");
+  divElements["h"].classList.add("container");
+  divElements["i"].classList.add("container");
+  divElements["j"].classList.add("container");
+
+  const subTree = Object.values(elm).map((x) => divElements[x]);
+
+  subTree.forEach((x) => {
+    if ((Array.from(divElements["f"].childNodes).includes(x))||(Array.from(divElements[6].childNodes).includes(x)) ){
```
Comment (Mr-MRF-Dev) [inline mahta/JS/JS DOM-manipulation/script.js:63]: divElements[`6`] ???

File: mahta/JS/JS DOM-manipulation/script.js [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+  elm["d"].classList.add("container");
+  elm["e"].classList.add("container");
+  elm["f"].classList.add("container");
+  elm["g"].classList.add("container");
+  elm["h"].classList.add("container");
+}
+
+const reset = document.querySelector("#reset");
+reset.addEventListener("click", firstPage);
+
```
Comment (Mr-MRF-Dev) [inline mahta/JS/JS DOM-manipulation/script.js:115]: Where is the reset ID for a button defined?

File: mahta/JS/JS DOM-manipulation/script.js [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+  const blue = Math.floor(Math.random() * (b - 100) + 100);
+  return `rgb(${yellow}, ${yellow}, ${blue})`;
+}
+
+function onButtonClickSet() {
+  document.body.innerHTML = "";
+  const buttonTag = document.createElement("button");
+  buttonTag.textContent = "Reset";
+  buttonTag.addEventListener("click", firstPage);
+  document.body.appendChild(buttonTag);
```
Comment (Mr-MRF-Dev) [inline mahta/JS/JS DOM-manipulation/script.js:12]: When you click the button, you should use DOM to change color or delete without resetting the page and creating new elements!

File: mahta/JS/JS DOM-manipulation/script.js [hunk 10]
```diff
@@ -0,0 +1,116 @@
...
+    c: 2,
+    d: 3,
+    f: 4,
+    g: 5,
+    h: 6,
+    i: 7,
+    j: 8,
+  };
+
+  const divElements = {};
```
Comment (Mr-MRF-Dev) [inline mahta/JS/JS DOM-manipulation/script.js:26]: Please make the variable a global variable.

**Review by Mr-MRF-Dev — CHANGES_REQUESTED** [review]

File: mahta/JS/JS DOM-manipulation/script.js [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+  const buttonTag = document.createElement("button");
+  buttonTag.textContent = "Reset";
+  buttonTag.addEventListener("click", firstPage);
+  document.body.appendChild(buttonTag);
+  Object.keys(elm).forEach((element) => {
+    elm[element] = document.createElement("div");
+    elm[element].textContent = element;
+  });
+  firstState();
+  dif();
```
Comment (Mr-MRF-Dev) [inline mahta/JS/JS DOM-manipulation/script.js:49]: It is not necessary to create a new node to change an element. Instead, follow the syntax to modify your elements.

```js
 // document.body.innerHTML = "";
  const buttonTag = document.getElementsByTagName("button")[0];
  buttonTag.textContent = "Reset";
  buttonTag.addEventListener("click", firstPage);
  // document.body.appendChild(buttonTag);
  Object.keys(elm).forEach((element) => {
   // elm[element] = document.createElement("div");
    const noode = document.createTextNode(element);
    elm[element].appendChild(noode);
  });
  // firstState();
  dif();

```

**Review by Mr-MRF-Dev — CHANGES_REQUESTED** [review]

**Review by Mr-MRF-Dev — DISMISSED** [review]

**Review by Mr-MRF-Dev — APPROVED** [review]

**Review by Mr-MRF-Dev — APPROVED** [review]

**Review by Mr-MRF-Dev — DISMISSED** [review]

**Review by Mr-MRF-Dev — DISMISSED** [review]
🌾

**Review by mkermani144 — CHANGES_REQUESTED** [review]
Many irrelevant changes are present in your PR. Please fix and request a review again.

**Pull request closed without merging**